src/screens/Home/Home.tsx

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from "react";
+import { View, Text, ActivityIndicator, Button } from "react-native";
+import AsyncStorageManager from "@storage/AsyncStorageManager";
+import styles from "./styles";
+
+const Home = ({ navigation: { replace } }: HomeScreenParamList) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(true);
+  const storageManager = AsyncStorageManager.getInstance();
+
+  useEffect(() => {
+    const fetchUserData = async () => {
+      const userId = await storageManager.getLoginUserId();
+
+      if (userId) {
+        const allUsers = storageManager.getUsers();
+        const loggedInUser = allUsers.find(user => user.id === userId);
+        setUser(loggedInUser || null);
+      }
+
+      setLoading(false);
+    };
+
+    fetchUserData();
+  }, []);
+
+  const handleLogout = async () => {
+    await storageManager.saveLoginUserId("");
+    replace("Login");
+  };
+
+  if (loading) {
+    return (
+      <View style={styles.loadingContainer}>
+        <ActivityIndicator
+          size="large"
+          color="#0000ff"
+        />
+      </View>
+    );
+  }
+
+  if (!user) {
```

Comment:
if the user is not present, how come the user got navigated to Home??

Reply:
This issue is been solved
I was also having problem with fetching `loginUserId` key from async storage.
It was signing successfully and storing the id of user but was unable to fetch.
So i add this case for testing purpose.

This home page is just a dummy to check authentication flow